packages/ts/react-routing/test/useRouteMetadata.spec.tsx

```diff
@@ -0,0 +1,94 @@
+import { expect } from '@esm-bundle/chai';
+import { render } from '@testing-library/react';
+import { createMemoryRouter, RouterProvider } from 'react-router-dom';
+import { type RouteObjectWithMetadata, useRouteMetadata, type RouteMetadata } from '../src';
+
+let metadata: RouteMetadata | undefined;
+
+const CaptureMetadata = () => {
+  metadata = useRouteMetadata();
+  return <div />;
+};
+
+const mainMetadata = {
+  title: 'Main route',
+  icon: 'main',
+  requireAuthentication: true,
+  rolesAllowed: ['ROLE_MAIN'],
+};
+const nestedMetadata = {
+  title: 'Nested route',
+  icon: 'nested',
+  requireAuthentication: true,
+  rolesAllowed: ['ROLE_NESTED'],
+};
+
+const routes: RouteObjectWithMetadata[] = [
+  {
+    path: '/main',
+    element: <CaptureMetadata />,
+    handle: mainMetadata,
```

Comment:
Why is it metadata is specified as `handle` and not e.g. `metadata`?

Reply:
The types and hook are just a glorified wrapper around the ones from `react-router-dom`. The `useMatches` hook used internally does not return the full route object but only some properties of it, one of which is `handle`. So if we add some custom property to the route itself we would not be able to retrieve it that way, but would have to come up with our own approach of accessing all routes (no default way to do that in `react-router-dom`), and then identifying the current one.